nixos/modules/services/torrent/deluge.nix

```diff
@@ -118,36 +118,78 @@ in {
             more informations.
           '';
         };
+
+        user = mkOption {
+          type = types.str;
+          default = "deluge";
+          description = ''
+            User account under which deluge runs.
+          '';
+        };
+
+        group = mkOption {
+          type = types.str;
+          default = "deluge";
+          description = ''
+            Group under which deluge runs.
+          '';
+        };
       };
 
       deluge.web = {
         enable = mkEnableOption "Deluge Web daemon";
+
+        user = mkOption {
+          type = types.str;
+          default = cfg.user;
+          description = ''
+            User account under which deluge web daemon runs.
+          '';
+        };
+
+        group = mkOption {
+          type = types.str;
+          default = cfg.group;
+          description = ''
+            Group under which deluge web daemon runs.
+          '';
+        };
+
         port = mkOption {
-        type = types.port;
+          type = types.port;
           default = 8112;
           description = ''
             Deluge web UI port.
           '';
         };
+
+        openFirewall = mkOption {
+          type = types.bool;
+          default = false;
+          description = ''
+            Open ports in the firewall for deluge web daemon
+          '';
+        };
       };
     };
   };
 
   config = mkIf cfg.enable {
 
-    systemd.tmpfiles.rules = [ "d '${configDir}' 0770 deluge deluge" ]
+    systemd.tmpfiles.rules = [ "d '${configDir}' 0770 ${cfg.user} ${cfg.group}" ]
     ++ optional (cfg.config ? "download_location")
-      "d '${cfg.config.download_location}' 0770 deluge deluge"
+      "d '${cfg.config.download_location}' 0770 ${cfg.user} ${cfg.group}"
     ++ optional (cfg.config ? "torrentfiles_location")
-      "d '${cfg.config.torrentfiles_location}' 0770 deluge deluge"
+      "d '${cfg.config.torrentfiles_location}' 0770 ${cfg.user} ${cfg.group}"
     ++ optional (cfg.config ? "move_completed_path")
-      "d '${cfg.config.move_completed_path}' 0770 deluge deluge";
+      "d '${cfg.config.move_completed_path}' 0770 ${cfg.user} ${cfg.group}";
 
     systemd.services.deluged = {
       after = [ "network.target" ];
       description = "Deluge BitTorrent Daemon";
       wantedBy = [ "multi-user.target" ];
-      path = [ pkgs.deluge ];
+      # Extraction utilities are required for the built-in "Extractor" plugin.
+      path = with pkgs; [ deluge unrar unzip gnutar xz p7zip bzip2 ];
```

Comment:
`unrar` is unfree, so I don't think it's a good idea to add this because then people need to `allowUnfree` just for that

Reply:
I've moved this list to an option, keeping the current packages (w/out `unrar`) as the default, so that users can add unfree packages if they want.

Reply:
I don't think we need an option for that. A default set of these packages without `unrar` seems fine. And if people do need `unrar` they can add it with `systemd.services.deluge.path = [ pkgs.unrar ]` themselves. Does that sound alright to you?

Reply:
I lean more towards having an option because I feel that would be more discoverable (via the website, when looking for deluge configuration options). 

I know that it wasn't immediately obvious to me, as a beginner, that I was able to override the `systemd.services.*` attributes set by a module. Also, if I ran into strange behaviour when using a Deluge plugin (due to a missing package that the plugin expected in the PATH), then I'd be more likely to look at options I might have missed first (and then see this option and identify the problem I have) than work out how to override the daemon arguments of Deluge to get more verbose logging, and then look at those logs to work out what is happening, and then work out how to override the path to provide the missing package.

Ultimately, I don't feel that strongly in either direction, so if you'd still prefer I remove that option then I'm happy to do so.

Reply:
How about using `optional (config.nixpkgs.config.allowUnfree or false) pkgs.unrar`? Then it will transparently work for people who do want to use unfree software. And the ones who intentionally keep `allowUnfree` disabled probably also wouldn't want to have unrar.

Reply:
I considered that when changing this to an option, but there are other built-in plugins included with Deluge (and users can load their own plugins), which may require other packages be available. I've only used the "Extractor" plugin myself, so I don't know what other plugins might expect. Providing users an easy way to add packages to Deluge's path would be useful to support these use-cases.

Reply:
I see, sounds good then